include/zenoh-pico/api/handlers.h

```diff
@@ -15,109 +15,74 @@
+    static inline void _z_##name##_elem_move(void *dst, const void *src) {                                        \
+        elem_move_f((recv_type *)dst, (const recv_type *)src);                                                    \
+    }                                                                                                             \
+    static inline void _z_##name##_send(const send_type *elem, void *context) {                                   \
+        void *internal_elem = elem_convert_f(elem);                                                               \
+        if (internal_elem == NULL) {                                                                              \
+            return;                                                                                               \
+        }                                                                                                         \
+        int8_t res = collection_push_f(internal_elem, context, _z_##name##_elem_free);                            \
+        if (res) {                                                                                                \
+            _Z_ERROR("%s failed: %i", #collection_push_f, res);                                                   \
+        }                                                                                                         \
+    }                                                                                                             \
+    static inline void _z_##name##_recv(recv_type *elem, void *context) {                                         \
+        int8_t res = collection_pull_f(elem, context, _z_##name##_elem_move);                                     \
+        if (res) {                                                                                                \
+            _Z_ERROR("%s failed: %i", #collection_pull_f, res);                                                   \
+        }                                                                                                         \
+    }                                                                                                             \
+                                                                                                                  \
+    static inline z_owned_##name##_t z_##name(size_t capacity) {                                                  \
+        z_owned_##name##_t channel;                                                                               \
+        channel.collection = collection_new_f(capacity);                                                          \
+        channel.send = z_##send_closure_name(_z_##name##_send, NULL, channel.collection);                         \
+        channel.recv = z_##recv_closure_name(_z_##name##_recv, NULL, channel.collection);                         \
```

Comment:
Should we have as well a `try_recv` method? It could be quite handy similar to what is being done in https://github.com/eclipse-zenoh/zenoh/pull/903/files.

It could come in a separate PR.